demo/web/index.html

```diff
@@ -0,0 +1,169 @@
+          const fileList = event.target.files;
+          readAudioFile(fileList[0], async (audioBuffer) => {
+            const f32PCM = audioBuffer.getChannelData(0);
+            const i16PCM = new Int16Array(f32PCM.length);
+
+            const INT16_MAX = 32767;
+            const INT16_MIN = -32768;
+            i16PCM.set(
+                    f32PCM.map((f) => {
+                      let i = Math.trunc(f * INT16_MAX);
+                      if (f > INT16_MAX) i = INT16_MAX;
+                      if (f < INT16_MIN) i = INT16_MIN;
+                      return i;
+                    })
+            );
+
+            writeMessage("Transcribing audio file...");
+            document.getElementById("result").innerHTML = await leopard.process(i16PCM, {transfer: true});
+            writeMessage("Transcribing audio file... done!");
+          });
+        });
+
+        const displayTimer = document.getElementById("displayTimer");
+        const recordButton = document.getElementById("recordAudio");
+        const stopRecord = document.getElementById("stopRecord");
+
+        let timer = null;
+        let webVp = null;
+        let currentTimer = 0.0;
+        let audioData = [];
+        recordButton.addEventListener("click", async () => {
+          displayTimer.style.display = "inline";
+          stopRecord.style.display = "inline";
+          recordButton.style.display = "none";
+
+          currentTimer = 0.0;
+          audioData = [];
+
+          try {
+            const worker = new Worker("worker.js");
+            webVp = await window.WebVoiceProcessor.WebVoiceProcessor.init({
+              engines: [worker],
+              start: true
+            });
+
+            writeMessage("Recording audio...");
+            worker.onmessage = (event) => {
+              audioData.push(event.data);
+            }
+            timer = setInterval(() => {
+              currentTimer += 0.1;
+              displayTimer.innerText = `${currentTimer.toFixed(1)} / 120`;
+              if (currentTimer === 120) {
+                stopRecord.click();
+              }
+            }, 100);
+          } catch (e) {
+            writeMessage(e);
+          }
+        });
+
+        stopRecord.addEventListener("click", async () => {
+          displayTimer.style.display = "none";
+          stopRecord.style.display = "none";
+          recordButton.style.display = "inline";
+
+          if (webVp !== null) {
+            webVp.stop();
+          }
+          clearInterval(timer);
+
+          const frames = new Int16Array(audioData.length * 512);
+          for (let i = 0; i < audioData.length; i++) {
+            frames.set(audioData[i], i * 512);
+          }
+
+          writeMessage("Transcribing audio file...");
+          document.getElementById("result").innerHTML = await leopard.process(frames, {transfer: true});
+          writeMessage("Transcribing audio file... done!");
+        });
+      }
+
+      function writeMessage(message) {
```

Comment:
would it make sense to move functions to the top with the other functions?

Reply:
the other functions are inside window.onload. this should be outside.